### DISCUSSION TITLE
Integrate chartbrew into subdirectory

### DISCUSSION THREAD

**Discussion by n84ck** [chartbrew/chartbrew#163] [Ideas]
Hello, Is there a possibility to integrate chartbrew into domain subdirectory?
Example: www.xyz.com -> normal webpage www.xyz/chartbrew -> chartbrew.

Currently the chartbrew want to serve the static files from the root of the domain.

**Comment by razvanilin** [comment]
Hey @n84ck, I haven't done it myself, but this might help: https://css-tricks.com/how-to-serve-a-subdomain-as-a-subdirectory/

So the main points:
1. Run Chartbrew normally on a subdomain first
2. Use one of the methods in the article above to redirect the subdomain to a subdirectory

I use Cloudflare for this kind of use case for Chartbrew's blog and it works wonders: https://chartbrew.com/blog

**Comment by n84ck** [comment]
Hi @razvanilin, I will try it, but currently I'm struggling with the official Docker image. I can't serve the website behind a domain, because the frontend. It send the requests to address http://localhost:4019. The REACT_APP_API_HOST variable doesn't help. Should I rebuild the docker image?
![image](https://user-images.githubusercontent.com/43924278/193471130-2b23675e-a2d5-46db-9b26-7385fdf4812f.png)

**Comment by razvanilin** [comment]
you will need to change the variable to your actual domain where you host your backend. Following your examples above, you could run the backend on a subdomain like `www.api.xyz.com`, then change the `REACT_APP_API_HOST` to `www.api.xyz.com`

**Comment by n84ck** [comment]
I tried with docker-compse as you can see on the attached image. The client still wants to connect to localhost. 

![image](https://user-images.githubusercontent.com/43924278/193883166-850c50f1-8027-4a04-922c-6c5243017fef.png)

**Comment by razvanilin** [comment]
Can you check if the client app has been rebuilt? The variable gets set only if the client is rebuilt using the new environmental variables. This command from the Dockerfile needs to run again: https://github.com/chartbrew/chartbrew/blob/master/Dockerfile#L10

**Comment by n84ck** [comment]
This image is pulled from your official docker container registry, there is no source code and building process: https://hub.docker.com/r/razvanilin/chartbrew
If I understood correctly, your official image is only usable for local host?

**Comment by razvanilin** [comment]
I just had a look over this and it seems the UI is not rebuilt when pulling from docker, which means the new variables are not updated in the frontend app. I'll update this in the next version, but for now you can rebuild the app manually like this:

``` sh
# replace 'your_container_name' with the name of your docker container where Chartbrew is running

docker exec -it -w /code/client your_container_name npm run build
```

**Comment by sprklinginfo** [comment]
@n84ck, new here and wondering if you are able to run it in a subdirectory eventually? I face the same problem  that I only have a subdomain on a VM server and have no control to create more subdomains. could you share the docker-compose file you are using if it works? Thanks!
